lib/network_config.ml

```diff
@@ -64,13 +65,27 @@ let read_management_conf () =
 		in
 		let phy_interface = {default_interface with persistent_i = true} in
 		let bridge_interface = {default_interface with ipv4_conf; ipv4_gateway; persistent_i = true} in
-		let bridge = {default_bridge with
-			bridge_mac = Some mac;
-			ports = [device, {default_port with interfaces = [device]}];
-			persistent_b = true
-		} in
-		{interface_config = [device, phy_interface; bridge_name, bridge_interface];
-			bridge_config = [bridge_name, bridge];
+		let interface_config, bridge_config =
+			let primary_bridge_conf = {default_bridge with
+                        	bridge_mac = Some mac;
+	                        ports = [device, {default_port with interfaces = [device]}];
+        	                persistent_b = true
+                	} in
+			if vlan <> "" && not (String.startswith "xenbr" bridge_name) && bridge_name <> "" then begin
```

Comment:
If we have a VLAN, then we _always_ need to choose a name for the parent bridge ourselves, because it won't be available anywhere. So why do we have this complex if-statement?